### PR TITLE
(feature) Validator Status Enum

### PR DESCRIPTION
**Pull request by shreifabdallah**
### What changed? Why?

- Validator status is an enum instead of a hardcoded string
- Changed tests throughout 

**UX Improvement**

- The addition of the `enum` enables IDEs to automatically recommend valid statuses, improving developer experience 

image

- In addition to supplying a string, you can also supply a `ValidatorStatus`, like so

image

**Testing notes**

```
> validators = await Validator.list("ethereum-holesky", "eth", ValidatorStatus.PROVISIONED);
[
  Validator {
    model: {
      asset_id: 'eth',
      details: [Object],
      network_id: 'ethereum-holesky',
      status: 'provisioned',
      validator_id: '0x84e9cf81f57d5d7956fd1daf797dc4765c28797829b951f7ce4caa76ce68030023d26ef39277ebb20d2efac44a89d009'
    }
  },
  Validator {
    model: {
      asset_id: 'eth',
      details: [Object],
      network_id: 'ethereum-holesky',
      status: 'provisioned',
      validator_id: '0x90206707fbc5a4d21db711613e9c6c903dbb37ef6707e6037321a297ae83eb9656adf3ba7d5953cc7a72df148421cec2'
    }
  }
]
```

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: src/coinbase/types.ts [hunk 1]
```diff
@@ -675,6 +675,22 @@ export enum SponsoredSendStatus {
   FAILED = "failed",
 }
 
+export enum ValidatorStatus {
+  Unknown = "unknown",
```
Comment (drohit-cb) [inline src/coinbase/types.ts:679]: I think the enums should be all caps.

**Review by drohit-cb — APPROVED** [review]

File: src/coinbase/types.ts [hunk 2]
```diff
@@ -675,6 +676,22 @@ export enum SponsoredSendStatus {
   FAILED = "failed",
 }
 
+export enum ValidatorStatus {
```
Comment (John-peterson-coinbase) [inline src/coinbase/types.ts:679]: Please add doc string

File: src/tests/validator_test.ts [hunk 3]
```diff
@@ -29,33 +31,99 @@ describe("Validator", () => {
...
+    testCases.forEach(({ input, expected }) => {
+      it(`should return ${expected} for ${input}`, () => {
+        const validatorModel = mockEthereumValidator("100", input, "0xpublic_key_1");
+        const validator = new Validator(validatorModel);
+        expect(validator.getStatus()).toBe(expected);
+      });
+    });
+  });
+
+  describe("getAPIValidatorStatus", () => {
```
Comment (John-peterson-coinbase) [inline src/tests/validator_test.ts:67]:
```suggestion
  describe("#getAPIValidatorStatus", () => {
```

File: src/tests/validator_test.ts [hunk 4]
```diff
@@ -29,33 +31,99 @@ describe("Validator", () => {
     });
   });
 
+  describe("getStatus", () => {
```
Comment (John-peterson-coinbase) [inline src/tests/validator_test.ts:34]:
```suggestion
  describe(".getStatus", () => {
```

**Review by John-peterson-coinbase — COMMENTED** [review]
Small documentation nits otherwise LGTM!

**Pull request merged**